core/src/proxy.ts

```diff
@@ -0,0 +1,13 @@
+import getContract from './contracts';
+
+export const createProxy = async (network: string, proxyOwnerAddress: string) => {
+    const proxyFactoryContract = await getContract(network, 'PROXY_FACTORY', true);
+    await proxyFactoryContract['build(address)'](proxyOwnerAddress);
```

Comment:
Wouldn't this already return `proxy` as [per the contract code](https://etherscan.io/address/0xa26e15c895efc0616177b7c1e7270a4c7d51c997#code):
```
function build(address owner) public returns (DSProxy proxy) 
```

Reply:
The return value of the function in solidity is only available on chain. Exceptions are view functions / constants. even if `build` returns the proxy, it is not a constant or a view = we have to extract it differently.

Reply:
Well, you're right, but according to https://ethereum.stackexchange.com/a/88122 there are better ways to do it anyway: Instead of building a new filter and fetching the last event, use executed transaction hash to fetch transaction receipt + events emitted with it. Maybe you can even make a helper from it? eg `getTransactionEvents`?